src/common/com/intellij/plugins/haxe/ide/refactoring/introduce/HaxeIntroduceHandler.java

```diff
@@ -449,13 +488,40 @@ protected void modifyDeclaration(@NotNull PsiElement declaration) {
   }
 
   @Nullable
-  protected HaxeReference createExpression(Project project, String name) {
-    return HaxeElementGenerator.createReferenceFromText(project, name);
+  protected HaxeReference createExpression(Project project, @NotNull HaxeIntroduceOperation operation) {
+    return HaxeElementGenerator.createReferenceFromText(project, operation.getName());
   }
 
   @Nullable
   protected PsiElement replaceExpression(PsiElement expression, PsiElement newExpression, HaxeIntroduceOperation operation) {
-    return expression.replace(newExpression);
+
+    PsiElement nextToken = findNextToken(expression, operation);
+    String nextText = null != nextToken ? nextToken.getText() : "";
+
+    PsiElement insertedElement = expression.replace(newExpression);
+
+    // If the element we are replacing ended with a semi-colon or was a block statement, the new expression should end with a colon.
+    String elementText = operation.getElement().getText();
+    char lastChar = elementText.charAt(elementText.length()-1);
+    if (lastChar == ';' || (lastChar == '}' && !";".equals(nextText)) ) {
+      insertedElement.addAfter(HaxeElementGenerator.createEmptyStatement(expression.getProject()), expression);
+    }
```

Comment:
The earlier fixup dealt with adding the semi-colon to the new variable declaration.  This one deals with adding it on each of the occurrences of the extracted statement.